webapp/loggedin/modules/home/homeView.js

```diff
@@ -19,77 +19,64 @@ import { homeModules } from './homeModules'
 
 import * as SurveyState from '../../../survey/surveyState'
 
-class HomeView extends React.Component {
+const HomeView = props => {
 
-  componentDidUpdate (prevProps) {
-    const { surveyInfo, history } = this.props
-    const { surveyInfo: prevSurveyInfo } = prevProps
+  const { surveyInfo } = props
+  const i18n = useI18n()
 
-    // active survey change
-    if (
-      surveyInfo && (
-        // new survey created
-        !prevSurveyInfo
-        // changed from survey list
-        || surveyInfo.id !== prevSurveyInfo.id
-      )
-    ) {
-      history.push(appModuleUri(homeModules.dashboard))
-    }
+  useEffect(() => {
+    const { history } = props
 
-    // survey deleted
-    if (Survey.isValid(prevSurveyInfo) && !Survey.isValid(surveyInfo)) {
+    if (Survey.isValid(surveyInfo)) {
+      // new survey created or active survey changed
+      history.push(appModuleUri(homeModules.dashboard))
+    } else {
+      // survey deleted
       history.push(appModuleUri(homeModules.surveyList))
     }
-  }
+  }, [surveyInfo])
```

Comment:
Should we use the survey uuid here? (it's maybe safer?)

Reply:
I would only made it a bit more compact, maybe using a ternary expression... e.g. history.push(appModules(Survey.isValid(surveyInfo) ? ...

Reply:
I meant the parameter to be passed to the effect